### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException when syncing data

### ISSUE THREAD

**Issue by ewa-radomska-codeclusive** [aws-amplify/amplify-flutter#3307]
### Description

Sometimes during syncing models we see this error:
```
java.lang.ArrayIndexOutOfBoundsException: length=49; index=49 at java.util.ArrayList.add(ArrayList.java:468) at com.amazonaws.amplify.amplify_datastore.DataStoreHubEventStreamHandler.getHubListener$lambda-0(Unknown Source:12) at com.amazonaws.amplify.amplify_datastore.DataStoreHubEventStreamHandler.a(Unknown Source:0) at d.b.a.e.s0.onEvent(Unknown Source:2) at com.amplifyframework.hub.AWSHubPlugin.a(Unknown Source:4) at d.c.g.b.run(Unknown Source:4) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637) at java.lang.Thread.run(Thread.java:1012)
```

It shows in both headless and foreground mode. It doesn't affect app working, but seems sometimes adding hub event to event history fails.

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [X] DataStore
- [ ] Notifications (Push)
- [ ] Storage

### Steps to Reproduce

Hard to reproduce it in the app with particular steps. One of our team's member found this when add object and than immediately edit it (but it's not showing all the time in this particular case). We see a lot of this bugs in production and everytime it shows during sync when a lot of models are syncing like in those logs:
```
[Headless HubListener] networks active : true
13:04:17
[Headless HubListener] sync on models : [Yepic, Image, ImageNote, Task, TaskNote, Stop, YepicVisit, OpenVisit, Note, YepicPlace, VisitNote]
13:04:17
[Headless HubListener] sync on model : Yepic  , added : 0, deleted : 0, updated : 0, delta sync : true, full sync : false
13:04:20
[Headless HubListener] sync on model : Image  , added : 0, deleted : 0, updated : 0, delta sync : true, full sync : false
13:04:20
[Headless HubListener] sync on model : ImageNote  , added : 0, deleted : 0, updated : 0, delta sync : true, full sync : false
13:04:20
[Headless HubListener] sync on model : Task  , added : 0, deleted : 0, updated : 0, delta sync : true, full sync : false
13:04:20
[Headless HubListener] sync on model : TaskNote  , added : 0, deleted : 0, updated : 0, delta sync : true, full sync : false
13:04:23
[Headless HubListener] sync on model : Stop  , added : 0, deleted : 0, updated : 0, delta sync : true, full sync : false
13:04:24
[Headless HubListener] sync on model : YepicVisit  , added : 0, deleted : 0, updated : 0, delta sync : true, full sync : false
13:04:24
[Headless HubListener] YepicVisit model synced - turning observer on
13:04:24
CRASH
java.lang.ArrayIndexOutOfBoundsException: length=73; index=73
MAPPINGS
Missing 
STATUS

Open
Received:2023-05-24 13:04:24
java.lang.ArrayIndexOutOfBoundsException: length=73; index=73
	at java.util.ArrayList.add(ArrayList.java:468)
	at com.amazonaws.amplify.amplify_datastore.DataStoreHubEventStreamHandler.getHubListener$lambda-0(Unknown Source:12)
	at com.amazonaws.amplify.amplify_datastore.DataStoreHubEventStreamHandler.a(Unknown Source:0)
	at d.b.a.e.s0.onEvent(Unknown Source:2)
	at com.amplifyframework.hub.AWSHubPlugin.a(Unknown Source:4)
	at d.c.g.b.run(Unknown Source:4)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
	at java.lang.Thread.run(Thread.java:1012)
Expand
13:04:24
[Headless HubListener] sync on model : OpenVisit  , added : 0, deleted : 0, updated : 0, delta sync : true, full sync : false
```

looks like everytime the problem is connected with adding new event to event history. We are trying to figure out if the problem is on our code side or on Amplify Flutter side and if we are doing something wrong what it can be. Any suggestions will be very helpful

### Platforms

- [ ] iOS
- [X] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.10.4

### Amplify Flutter Version

1.1.0

### Deployment Method

Amplify CLI

**Comment by HuiSF** [comment]
Hi @stevegaunt thanks for the diagnose, we will look into this issue.

**Comment by stevegaunt** [comment]
@HuiSF @NikaHsn 
any updates with this??  It's effecting live users for us

**Comment by NikaHsn** [comment]
@stevegaunt I don't have an update yet. I'll see how we can prioritize this and will update here on progress. thanks for your patience.

**Comment by NikaHsn** [comment]
This has been fixed in v1.3.1. Thanks for your patience resolving this issue.
